### PR TITLE
Posts list - post upload trash/delete confirm dialog

### LINKED ISSUE #6306: Add dialog to confirm cancelling media uploads when deleting post
Per #6297 we decided to prompt the user to confirm their decision to trash a post with in-progress media uploads.

Rather than toasting we will show a dialog that asks them if they want to cancel media uploads and delete the post.

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #6306 

To test:
1. Start a new draft
2. include some media in it (preferrably some big images to make sure it takes some time to upload)
3. exit the editor
4. check the upload is progressing in the Posts list and the notifications
5. tap on "trash/delete" on the post being uploaded in the post list
6. the dialog is shown

screenshot_20170725-103137

7. tap on OK
8. You'll see the post is hidden and the toast "Post deleted - UNDO" is shown.
9. wait a few seconds and verify the snackbar is gone and the notification is cleared as well when the snackbar is hidden.

cc @nbradbury @aforcier

### CODE REVIEW

**Comment by aforcier** [comment]
I think the dialog should also have a 'cancel' option - but let's get some design input on this in general.

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java [hunk 1]
```diff
@@ -318,6 +318,20 @@ public static boolean isPostUploading(PostModel post) {
         return post != null && PostUploadHandler.isPostUploading(post);
     }
 
+    public static void abortPostAndRelatedMediaUpload(PostModel post) {
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: Could we refactor this to call `cancelQueuedPostUpload()` internally and drop some code duplication?

Also, what do you think of calling it `cancelQueuedPostUploadAndRelatedMedia()`, for a bit more consistency with that (similarly-behaving) method?

**Comment by iamthomasbishop** [comment]
> I think the dialog should also have a 'cancel' option

Agreed. Perhaps the copy should probably read something like:

### Delete post?

Media in this post is currently uploading. Are you sure you'd like to delete this post?

[cancel] [delete]

Maybe the body copy could be simplified, but I'd defer to Editorial.

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java (hunk 1 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: addressed in ee8e0b0

**Comment by mzorz** [comment]
Added the `cancel` button and changed `OK` for `DELETE` as per @iamthomasbishop 's suggestion in a9d4256

screenshot_20170727-145222


hi @iamthomasbishop! re: this:
> Media in this post is currently uploading. Are you sure you want to delete this post?

I'd prefer something more clear as this assertion doesn't imply the media will _also_ be cancelled (along with the post being deleted), as it's currently suggested in this PR. In this sense I'd stick with the current message, but waiting for Editorial seems a good choice 👍

**Comment by michelleweber** [comment]
I think we can nix the bulky word "related," as it's pretty clear without it (and the user can see the media upload bar). I suggest:

Are you sure? Deleting this post will also cancel the media upload.

**Comment by mzorz** [comment]
Awesome! Thank you @iamthomasbishop and @michelleweber 🙇  addressed this change in a7943b0

Ready to go @aforcier

**Comment by aforcier** [comment]
Looks good! :shipit:

**Pull request merged**